subsys/net/lib/lwm2m/Kconfig

```diff
@@ -162,7 +162,14 @@ config LWM2M_PEER_PORT
 	depends on LWM2M_RD_CLIENT_SUPPORT
 	default 5683
 	help
-	  This is the default server port to connect to for LWM2M communication
+	  This is the default server port to connect to for LWM2M communication.
+
+config LWM2M_FIRMWARE_PORT
+	int "LWM2M firmware server port"
+	depends on LWM2M_RD_CLIENT_SUPPORT
```

Comment:
I don't see any relevance to `LWM2M_RD_CLIENT_SUPPORT`, this config shouldn't depend on it. I'd rather see `LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT` in this place, or even none to prevent unnecessary ifdef slicing of the code.

Reply:
Removed the dependency